src/parcsr_ls/par_mgr_solve.c

```diff
@@ -220,30 +226,21 @@ hypre_MGRSolve( void               *mgr_vdata,
       {
          old_resnorm = resnorm;
 
-         if ( logging > 1 )
+         if (logging > 1)
          {
             hypre_ParVectorCopy(F_array[0], residual);
-            hypre_ParCSRMatrixMatvec(alpha, A_array[0], U_array[0], beta, residual );
-            resnorm = hypre_sqrt(hypre_ParVectorInnerProd( residual, residual ));
+            hypre_ParCSRMatrixMatvec(fp_neg_one, A_array[0], U_array[0], fp_one, residual);
+            resnorm = hypre_sqrt(hypre_ParVectorInnerProd(residual, residual));
          }
          else
          {
             hypre_ParVectorCopy(F_array[0], Vtemp);
-            hypre_ParCSRMatrixMatvec(alpha, A_array[0], U_array[0], beta, Vtemp);
+            hypre_ParCSRMatrixMatvec(fp_neg_one, A_array[0], U_array[0], fp_one, Vtemp);
             resnorm = hypre_sqrt(hypre_ParVectorInnerProd(Vtemp, Vtemp));
          }
 
-         if (old_resnorm) { conv_factor = resnorm / old_resnorm; }
-         else { conv_factor = resnorm; }
-         if (rhs_norm > HYPRE_REAL_EPSILON)
-         {
-            rel_resnorm = resnorm / rhs_norm;
-         }
-         else
-         {
-            rel_resnorm = resnorm;
-         }
-
+         conv_factor = (old_resnorm > HYPRE_REAL_EPSILON) ? resnorm / old_resnorm : resnorm;
```

Comment:
This meaning has changed, but I think it is OK. Should the change be consistent with line 972 then?

Reply:
Yes, it should! Thanks!